app/scripts/lib/channels/senders/web-channel.js

```diff
@@ -54,20 +66,57 @@ define(function (require, exports, module) {
     }
   };
 
-  function createEvent(win, webChannelId, command, data, messageId) {
+  /**
+   * Create a WebChannelMessageToChrome event with the given `eventDetail`
+   *
+   * @param {Object} win recipient of the WebChannel event.
+   * @param {Object} eventDetail `detail` property of the event.
+   * @returns {Event} WebChannelMessageToChrome event.
+   */
+  function createEvent(win, eventDetail) {
     return new win.CustomEvent('WebChannelMessageToChrome', {
-      detail: {
-        id: webChannelId,
-        message: {
-          command: command,
-          data: data,
-          messageId: messageId
-        }
-      }
+      detail: formatEventDetail(win, eventDetail)
     });
   }
 
-  module.exports = WebChannelSender;
-});
+  /**
+   * Create an object for the `detail` field of a WebChannelMessageToChrome event.
+   *
+   * @param {String} webChannelId ID of the receiving WebChannel
+   * @param {String} command command name
+   * @param {Object} data payload
+   * @param {String} messageId messageId browser will respond with.
+   * @returns {Object}
+   */
+  function createEventDetail(webChannelId, command, data, messageId) {
+    return {
+      id: webChannelId,
+      message: {
+        command,
+        data,
+        messageId
+      }
+    };
+  }
 
+  /**
+   * Format the `detail` payload for the current browser.
+   *
+   * @param {Object} win receiving window
+   * @param {Object} eventDetail `detail` payload
+   * @returns {String|Object}
+   */
+  function formatEventDetail(win, eventDetail) {
+    const userAgent = new UserAgent(win.navigator.userAgent);
+    // Firefox Desktop >= 50 expects the detail to be sent as a string.
+    // See https://bugzilla.mozilla.org/show_bug.cgi?id=1275616 and
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=1238128
+    if (userAgent.isFirefoxDesktop() && userAgent.browser.version >= 50) {
```

Comment:
As we mentioned in bugzilla, another alternative would be to just wait a few months and lose the UA check completely :)